Eplant/views/ChromosomeViewer/ChromosomeView.tsx

```diff
@@ -0,0 +1,138 @@
+  ChromosomeViewerState,
+  ChromosomeViewerStateScheme,
+  Transform,
+} from './types'
+import ZoomControls from './ZoomControls'
+import { ChromosomeViewerObject } from '.'
+
+export const ChromosomeView = () => {
+  const { geneticElement } = useOutletContext<ViewContext>()
+  const { state, setState, initializeState } =
+    useURLState<ChromosomeViewerState>()
+  const [loadAmount, setLoadAmount] = useState(0)
+
+  const spaceRef = useRef<Space | null>(null)
+  const [messageOpen, setMessageOpen] = useState(true)
+  const handleClose = () => {
+    setMessageOpen(false)
+  }
+  const { data, isLoading, isError, error } = useQuery<ChromosomeViewerData>({
+    queryKey: [`chromosome`],
+    queryFn: async () => {
+      return ChromosomeViewLoader(geneticElement, setLoadAmount)
+    },
+  })
+
+  useEffect(() => {
+    // On mount, set the active actions and initialize the state
+    initializeState(ChromosomeViewerStateScheme)
+  }, [])
+
+  if ((isLoading && loadAmount < 100) || isError) {
+    return (
+      <LoadingPage
+        loadingAmount={loadAmount}
+        gene={geneticElement}
+        view={ChromosomeViewerObject}
+        error={ViewDataError.FAILED_TO_LOAD}
+      ></LoadingPage>
+    )
+  } else if (!data || !state) return <></>
+
+  return (
+    <Box>
+      {/* ZOOM CONTROLS */}
+      <ZoomControls spaceRef={spaceRef} scale={state.transform.dZoom} />
+      {/* CHROMOSOME VIEWER */}
+      <Space
+        ref={spaceRef}
+        onCreate={(vp) => {
+          vp.camera.recenter(
+            state.transform.dx,
+            state.transform.dy,
+            state.transform.dZoom
+          )
+          vp.setBounds({
+            x: [-650, 1300],
+            y: [-450, 815],
+            zoom: [0.05, 1000 - 0.3],
+          })
+        }}
+        onUpdated={(vp) => {
+          const transform: Transform = {
+            dx: vp.centerX,
+            dy: vp.centerY,
+            dZoom: vp.zoomFactor,
+          }
+          setState({
+            ...state,
+            transform: transform,
+          })
+        }}
+      >
+        <ChromosomeViewer
+          chromosomes={data.viewData}
+          scale={state.transform.dZoom}
+          annotations={[]}
+        />
+      </Space>
+      <Snackbar
+        id='chromosomeViewer_geneAnnotationMessage'
+        open={messageOpen}
+        autoHideDuration={1000}
+        onClose={handleClose}
+        sx={{ position: 'absolute' }}
+      >
+        <SnackbarContent
+          sx={(theme) => ({
+            backgroundColor: theme.palette.background.transparentOverlay,
+            color: theme.palette.secondary.contrastText,
+          })}
+          message={
+            <span>
+              Please allow a brief moment for gene annotations to be processed
+              &nbsp;
+              <CircularProgress size={12} />
+            </span>
+          }
+        />
+      </Snackbar>
+    </Box>
+  )
+}
+
+const ChromosomeViewLoader = async (
```

Comment:
See above comment about making this throw a ViewDataError on un-successful response